index.js

```diff
@@ -165,9 +166,15 @@ module.exports = function (match, command, args, opts) {
       start({ event: '', changed: '' })
     }
 
+    var mustListenToAllEvents = eventsToListen.indexOf('all') >= 0;
+
     // For any change, creation or deletion, try to run.
     // Restart if the last run is still active.
     watcher.on('all', function (event, changed) {
+      if (!mustListenToAllEvents && eventsToListen.indexOf(event) === -1) {
```

Comment:
I'm also ok with `if (filter.length && filter.indexOf(event) === -1) return`. I might just make this quick change since I'm the one who's been delaying you here, thanks for the PR!